### DISCUSSION TITLE
How to get Annotation data from Namespace in promql

### DISCUSSION THREAD

**Discussion by subasathees** [prometheus/prometheus#13978] [Q&A]
### What did you do?

We would like to add namespace description on our grafanan dashboard for each namespace, for that we have choosan annotation method and we are able to annotate the namespace description in kubernetes, but unable to view the annotation data in grafana (promql).

### What did you expect to see?

We would like to add namespace description on our grafanan dashboard for each namespace.
namespace Annotation data should show on grafana.

### What did you see instead? Under which circumstances?

namespace Annotation data not available on any matrix

### System information

Linux 3.10.0-1160.99.1.el7.x86_64 x86_64

**Comment by machine424** [comment]
This really lacks details, and I think this is not a problem with Prometheus: if you tell Prometheus to scrape a target exposing that metric with the annotation it will do it.

But it's up to you to provide that metric, are you using https://github.com/kubernetes/kube-state-metrics?

**Comment by machine424** [comment]
Let's move this into a discussion, don't hesitate to reach out to the community on slack fo Prometheus related support requests.
